go/flow/converge.go

```diff
@@ -164,6 +165,50 @@ func MapShardToSplit(task pf.Task, shards []pc.ListResponse_Shard, splitOnKey bo
 	return []pf.LabelSet{lhs, rhs}, nil
 }
 
+// MapPartitionToSplit maps a single partition journal contained in |journals|
+// to a desired split state, where the partition is evenly subdivided into
+// |splits| sub-partitions. |splits| must be a power-of-two.
+func MapPartitionToSplit(collection *pf.CollectionSpec, journals []pb.ListResponse_Journal, splits uint) ([]pf.LabelSet, error) {
+	if len(journals) != 1 {
+		return nil, fmt.Errorf("expected exactly one journal in the response")
+	} else if splits < 2 || bits.OnesCount(splits) != 1 {
```

Comment:
Should there be an upper bound on `splits`? Possibly checked down below after `range_` is computed, because the obvious sanity check would be that you have fewer splits than there are possible keys?

Reply:
Done